### PR TITLE
Fix Firefox textarea autoscroll

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Type a bunch of text into the quiet custom height TextArea in the "custom height with label" story, the cursor should always remain in view. Sanity check the other TextArea combos as well

## 🧢 Your Project:

RSP

### CODE REVIEW

**Review by snowystinger — DISMISSED** [review]
looks good for FF

do we want to try to solve the flickering? i was thinking a bit more about it, something like this

onBeforeInput => is max height already (has scrollbars before input)
onInput => is now max height (has scrollbars again)
if both of those are true, don't do the height calculation
else do it
reset hasscrollbarsbefore/after

File: packages/@react-spectrum/textfield/src/TextArea.tsx [hunk 1]
```diff
@@ -48,13 +48,19 @@ function TextArea(props: SpectrumTextFieldProps, ref: RefObject<TextFieldRef>) {
       input.style.height = `${input.scrollHeight + (input.offsetHeight - input.clientHeight)}px`;
       input.style.overflow = prevOverflow;
       input.style.alignSelf = prevAlignment;
+      // Fixes Firefox autoscrolling issue. Force scroll position to end of text area if user is adding to the end of the text area content
+      if (inputValue?.startsWith(lastInputValue.current)) {
+        input.scrollTop = input.scrollHeight;
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/textfield/src/TextArea.tsx:53]: it shouldn't, this only does it if we've appended, not inserted

[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-spectrum/textfield/src/TextArea.tsx:53]: try it! it's broken 😄

[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/textfield/src/TextArea.tsx:53]: It's not broken for me, can you specify the story and browser it's broken for you on? and what you're typing?

I could see this being broken if we typed "hello hello" and then inserted another "hello" in the middle. @LFDanLu probably better to do it the other way we discussed and check if the cursor position is at the end already

[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-spectrum/textfield/src/TextArea.tsx:53]: yeah I just inserted a bunch of newlines, then scrolled back up about half way and inserted another one.

File: packages/@react-spectrum/textfield/src/TextArea.tsx [hunk 2]
```diff
@@ -41,15 +40,25 @@ function TextArea(props: SpectrumTextFieldProps, ref: RefObject<TextFieldRef>) {
       let input = inputRef.current;
       let prevAlignment = input.style.alignSelf;
       let prevOverflow = input.style.overflow;
+      // Firefox scroll position is lost when overflow: 'hidden' is applied so use
+      // scrollbar-width: 'none' instead.
+      let prevSBWidth = CSS && CSS.supports && CSS.supports('scrollbar-width: none') && (input.style['scrollbar-width'] || 'auto');
```
Comment (devongovett) [inline packages/@react-spectrum/textfield/src/TextArea.tsx:45]:
```suggestion
      let prevSBWidth = typeof CSS !== 'undefined' && CSS.supports && CSS.supports('scrollbar-width: none') && (input.style['scrollbar-width'] || 'auto');
```

**Review by devongovett — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**